src/main/scala/tile/RocketTile.scala

```diff
@@ -154,16 +154,13 @@ class RocketTileModuleImp(outer: RocketTile) extends BaseTileModuleImp(outer)
     beu.module.io.errors.icache := outer.frontend.module.io.errors
   }
 
-  // Pass through various external constants and reports
+  // Pass through various external constants and reports that were bundle-bridged into the tile
   outer.traceSourceNode.bundle <> core.io.trace
   core.io.traceStall := outer.traceAuxSinkNode.bundle.stall
   outer.bpwatchSourceNode.bundle <> core.io.bpwatch
-  outer.frontend.module.io.reset_vector := constants.reset_vector
-
-  def regHart(x: UInt): UInt = if (p(HartPrefixKey)) RegNext(x) else x
-  core.io.hartid := regHart(constants.hartid)
-  outer.dcache.module.io.hartid := regHart(constants.hartid)
-  outer.frontend.module.io.hartid := regHart(constants.hartid)
+  core.io.hartid := outer.hartIdSinkNode.bundle
+  require(core.io.hartid.getWidth >= outer.hartIdSinkNode.bundle.getWidth,
+    s"core hartid wire (${core.io.hartid.getWidth}b) truncates external hartid wire (${outer.hartIdSinkNode.bundle.getWidth}b)")
```

Comment:
ok I see that this is where the width check is happening to make sure it doesnt get truncated, but it seems like people hooking up to tileHartIdNexusNode can do as they please and it won't be checked until later, wheras we COULD have checked the widths of the wires in the tileHartIdNexusNode to ensure none of them were > p(MaxHartIdBits)?

Reply:
The nexus node internally contains usages of `chisel3.DataMirror` that ensures that all the bundles on input and output edges are identical widths to one another, but you are right that it doesn't check against `p(MaxHartIdBits)`. But since now only the core is using ``p(MaxHartIdBits)`, it seems right to me that only the core checks against ``p(MaxHartIdBits)`? Like if we changed it to be part of `CoreParams` or `TileParams`, why should we have to change code in `subsystem` package, ideally?

Reply:
yeah, I agree with this. This is the place that cares so it seems right that this is the place that checks.

Reply:
Actually looking at the code again, I'm currently supply `0.U(p(MaxHartIdBits)` as the default value, which is treated as an input for the purposes of the `DataMirror` equivalency check. So it's already handled. I'll leave the more specific check in the core code though.